dialogs/SmoothingGroups.cpp

```diff
@@ -0,0 +1,82 @@
+#include "dialogs/SmoothingGroups.h"
+
+#include <QGroupBox>
+#include <QGridLayout>
+#include <QPushButton>
+#include <QLabel>
+
+using namespace ui;
+CSmoothingGroups::CSmoothingGroups( QWidget *pParent ) :
+	QDialog( pParent )
+{
+	this->setWindowTitle( tr( "Smoothing Groups" ) );
+
+	auto pDialogLayout = new QVBoxLayout( this );
+
+	// Smooth
+	auto pSmoothGroup = new QGroupBox( tr( "Smooth" ) );
+	auto pSmoothLayout = new QGridLayout( pSmoothGroup );
+	pSmoothLayout->setVerticalSpacing( 0 );
+	pSmoothLayout->setHorizontalSpacing( 0 );
+
+	int row = 0;
+	for ( int i = 1; i < 25; i++ )
+	{
+		auto pButton = new QPushButton( QString::number( i ), this );
+		pButton->setMaximumWidth( 25 );
+		pButton->setMaximumHeight( 20 );
+		connect( pButton, &QPushButton::released, [this, i]
+				 {
+					 onSmoothPressed( i );
+				 } );
+
+		// Add to the dialog
+		pSmoothLayout->addWidget( pButton, row, ( i - 1 ) % 8 );
+
+		// Increment the row
+		if ( i % 8 == 0 )
+			row++;
+	}
+
+	pDialogLayout->addWidget( pSmoothGroup );
+
+	// Hard
+	row = 0;
+	auto pHardGroup = new QGroupBox( tr( "Hard" ) );
```

Comment:
Needs to have `this` as the parent in the constructor

Reply:
Ah, alright, completely didn't notice these! I'll fix that now!